onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -2074,16 +2074,30 @@ def aten_convolution(
 ) -> TFloat:
     """convolution(Tensor input, Tensor weight, Tensor? bias, int[] stride, SymInt[] padding, int[] dilation, bool transposed, SymInt[] output_padding, int groups) -> Tensor"""
 
+    # 1D: rank=3
+    # 2D: rank=4
+    # 3D: rank=5
+    rank = len(input.shape)
+
+    rank_to_image_d = {3: 1, 4: 2, 5: 3}
+    image_d = rank_to_image_d[rank]
+
     if not isinstance(padding, Sequence):
-        padding = (padding, padding)
+        padding = [padding] * image_d
+    elif len(padding) != image_d:
```

Comment:
Assert len(padding) == 1 in this case?

Reply:
I used len(padding) > 0. I think this way makes more sense?

Reply:
What happens when len(padding) == 2, for example, when imsage_d == 3; is it a valid case? Down below we take padding[0] which implies all elements after the first element will be ignored.

Reply:
Yeah, I take the first one and expand it for the users, because we don't know what aten op would get.

Reply:
Do we want to raise an explicit error when len is not 1 then, because otherwise there will be silent errors?

Reply:
Done. Change it to only support len == 1